network/p2pNetwork_test.go

```diff
@@ -1154,3 +1154,141 @@ func TestMergeP2PAddrInfoResolvedAddresses(t *testing.T) {
 		})
 	}
 }
+
+// TestP2PEnableGossipService_NodeDisable ensures that a node with EnableGossipService=false
+// still can participate in the network by sending and receiving messages.
+func TestP2PEnableGossipService_NodeDisable(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	log := logging.TestingLog(t)
+
+	// prepare configs
+	cfg := config.GetDefaultLocal()
+	cfg.DNSBootstrapID = "" // disable DNS lookups since the test uses phonebook addresses
+
+	relayCfg := cfg
+	relayCfg.NetAddress = "127.0.0.1:0"
+
+	nodeCfg := cfg
+	nodeCfg.EnableGossipService = false
+	nodeCfg2 := nodeCfg
```

Comment:
Apologies if I'm misunderstanding the nuance of this test, but doesn't this result in both nodes having `EnableGossipService` disabled?

Reply:
There are two tests with two nodes: one with relayCfg and other with nodeCfg/nodeCfg2 config, and a node with nodeCfg/nodeCfg2 does have EnableGossipService=false